MinimalLifts.m

```diff
@@ -0,0 +1,231 @@
+MaximalOrder:=function(G)
+  x:=[z:z in Generators(G)];
+a:=x[1];
+n:=Order(G);
+i:=1;
+while Order(a) ne n do
+	   i:=i+1;
+a:=a*x[i];
+end while;
+return a;
+end function;
+
+
+CentralProductPC:=function(n,H,Z)
+  T:=H meet Z;
+k1:=Order(T);
+if Order(Z) eq 1 then
+return H;
+end if;
+p:=Order(H);
+k2:=n*k1;
+k:=k2/p;
```

Comment:
Can use `k2 div p` to get an integer immediately.  It will give the same result as `Numerator(k2/p)` when `p` divides `k2`, and the floor of `k2/p` if not.